tests/test_drs.py

```diff
@@ -0,0 +1,176 @@
+    }
+    responses.add(responses.GET, full_url, json=presigned_url, status=status)
+    return presigned_url
+
+
+def get_doc(
+    has_version=True, multiple_endpointurl=False, has_content=False, urls=list()
+):
+    doc = {
+        "form": "object",
+        "size": 123,
+        "urls": ["s3://endpointurl/bucket/key"],
+        "hashes": {"md5": "8b9942cf415384b27cadf1f4d2d682e5"},
+    }
+    if has_version:
+        doc["version"] = "1"
+    if multiple_endpointurl:
+        for url_type in ["gs", "ftp", "sftp"]:
+            doc["urls"].append("{}://endpointurl/bucket/key".format(url_type))
+
+    return doc
+
+
+def test_drs_get(client, user):
+    data = get_doc()
+    res_1 = client.post("/index/", json=data, headers=user)
+    assert res_1.status_code == 200
+    rec_1 = res_1.json
+    res_2 = client.get("/ga4gh/drs/v1/objects/" + rec_1["did"])
+
+    assert res_2.status_code == 200
+    rec_2 = res_2.json
+    assert rec_2["id"] == rec_1["did"]
+    assert rec_2["size"] == data["size"]
+    for k in data["hashes"]:
+        assert rec_2["checksums"][0]["checksum"] == data["hashes"][k]
+        assert rec_2["checksums"][0]["type"] == k
+    assert rec_2["version"]
+    assert rec_2["self_uri"] == "drs://fictitious-commons.io/" + rec_1["did"]
+
+
+def test_drs_multiple_endpointurl(client, user):
+    data = get_doc(multiple_endpointurl=True)
+
+    res_1 = client.post("/index/", json=data, headers=user)
+    assert res_1.status_code == 200
+    rec_1 = res_1.json
+    res_2 = client.get("/ga4gh/drs/v1/objects/" + rec_1["did"])
+
+    assert res_2.status_code == 200
+    rec_2 = res_2.json
+    assert rec_2["id"] == rec_1["did"]
+
+    object_urls = {
+        "sftp": "sftp://endpointurl/bucket/key",
+        "ftp": "ftp://endpointurl/bucket/key",
+        "gs": "gs://endpointurl/bucket/key",
+        "s3": "s3://endpointurl/bucket/key",
+    }
+    data = get_doc(urls=list(object_urls.values()))
+    for url in rec_2["access_methods"]:
+        protocol = url["type"]
+        assert url["access_url"]["url"] == object_urls[protocol]
+
+
+@responses.activate
+def test_drs_get_with_presigned_url(client, user):
+    data = get_doc()
+    res_1 = client.post("/index/", json=data, headers=user)
+    assert res_1.status_code == 200
+    rec_1 = res_1.json
+    presigned = generate_presigned_url_response(rec_1["did"])
+    res_2 = client.get(
+        "/ga4gh/drs/v1/objects/" + rec_1["did"], headers={"AUTHORIZATION": "12345"}
+    )
+    assert res_2.status_code == 200
+    rec_2 = res_2.json
+    assert rec_2["id"] == rec_1["did"]
+    assert rec_2["size"] == data["size"]
+    for k in data["hashes"]:
+        assert rec_2["checksums"][0]["checksum"] == data["hashes"][k]
+        assert rec_2["checksums"][0]["type"] == k
+
+    assert rec_2["access_methods"][0]["access_url"] == presigned
+
+
+def test_drs_list(client, user):
```

Comment:
this test is fine but it doesn't really test the listing functionality since there is only 1 record